### PR TITLE
[AIRFLOW-6593] Add GCP Stackdriver Alerting Hooks and Operators

### PR DESCRIPTION
**Pull request by jithin-sukumar**
### Description
Add GCP Stackdriver Alerting Hooks and Operators

---
Issue link: [AIRFLOW-6593](https://issues.apache.org/jira/browse/AIRFLOW-6593)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/stackdriver.py [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+    def list_alert_policies(
+        self,
+        project_id=None,
+        format_=None,
+        filter_=None,
+        order_by=None,
+        page_size=None,
+        retry=DEFAULT,
+        timeout=DEFAULT,
+        metadata=None
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/stackdriver.py]: Would you mind adding type annotations? :)

File: airflow/providers/google/cloud/operators/stackdriver.py [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+        self.gcp_conn_id = gcp_conn_id
+        self.project_id = project_id
+        self.delegate_to = delegate_to
+        self.hook = None
+
+    def execute(self, context):
+        if self.hook is None:
+            self.hook = StackdriverHook(gcp_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to)
+
+        return self.hook.list_alert_policies(
```
Comment (turbaszek) [inline airflow/providers/google/cloud/operators/stackdriver.py:113]: Would it be worth to add some info logging before and after hook call?

File: airflow/providers/google/cloud/hooks/stackdriver.py (hunk 1 above)
Comment (jithin-sukumar) [reply airflow/providers/google/cloud/hooks/stackdriver.py]: how about https://github.com/apache/airflow/pull/7322/commits/86898011b3a31e03830d56af567193a6e6d54a2c

File: airflow/providers/google/cloud/operators/stackdriver.py (hunk 2 above)
Comment (jithin-sukumar) [reply airflow/providers/google/cloud/operators/stackdriver.py:113]: please check https://github.com/apache/airflow/pull/7322/commits/86898011b3a31e03830d56af567193a6e6d54a2c

**Review by turbaszek — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]
Thanks!

**Pull request merged**